modules/javafx.graphics/src/main/java/com/sun/glass/ui/mac/MacRobot.java

```diff
@@ -24,75 +24,133 @@
  */
 package com.sun.glass.ui.mac;
 
-import com.sun.glass.ui.*;
+import javafx.scene.image.WritableImage;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.paint.Color;
+
+import com.sun.glass.ui.Application;
+import com.sun.glass.ui.GlassRobot;
+import com.sun.glass.ui.Pixels;
 
 /**
  * MacOSX platform implementation class for Robot.
  */
-final class MacRobot extends Robot {
+final class MacRobot extends GlassRobot {
 
     // TODO: get rid of native Robot object
     private long ptr;
 
     private native long _init();
-    @Override protected void _create() {
+    @Override
+    public void create() {
+        Application.checkEventThread();
         ptr = _init();
     }
 
-    private native void _destroy(long ptr);
-    @Override protected void _destroy() {
+    native private void _destroy(long ptr);
+    @Override
+    public void destroy() {
+        Application.checkEventThread();
         if (ptr == 0) {
             return;
         }
         _destroy(ptr);
```

Comment:
When the pointer will be set to "0"?

Reply:
This looks like a bug to me. I would think that it should be set to null right after the call to _destroy (as is done in the IosRobot code).

Reply:
As a follow-on: I note that there is no code that actually calls destroy. As a follow-up issue, we should either call this via a disposer mechanism (if it is needed to clean up native resources), or remove the method entirely.

Reply:
Yes, It will be good to fix it soon, because it will cause a native memory leak.